### PR TITLE
Fix #2631: A11y Options Language screen

### LINKED ISSUE #2631: [A11y] Options Language Screen
## Current Output

https://user-images.githubusercontent.com/9396084/107033302-1a334b00-67db-11eb-94aa-6a77dd318e78.mp4

## Issues Identified
- [ ] The talkback reads this screen as **Oppia** at start, instead it should be either **Default Audio Language** or **App Language** 
- [ ] When any particular item is selected the output should be : **French, checked, 2 of 4** or **French, 2 of 4, checked** and checkbox should bot be selected separately. This will reduce one step for the learner.
Note: To understand the above issue(s) completely, it is recommended that you setup Talkback and play with the app keeping it on and that will give you better context.

**Accessibility Guide**: https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide

### PR DESCRIPTION
**Pull request by abhinav12k**
## Explanation
Fixes #2631 

## Current Output

https://user-images.githubusercontent.com/51455561/109333414-72fb7e00-7885-11eb-9025-1dd5409deda5.mp4

## Checklist
- [x] The talkback reads this screen as **Oppia** at start, instead it should be either **Default Audio Language** or **App Language**
- [x] When any particular item is selected the output should be : French, checked, 2 of 4 or French, 2 of 4, checked and checkbox should bot be selected separately. This will reduce one step for the learner.
    Note: To understand the above issue(s) completely, it is recommended that you setup Talkback and play with the app keeping it on and that will give you better context.

**Accessibility Guide**: https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
 <project version="4">
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" default="true" project-jdk-name="1.8" project-jdk-type="JavaSDK">
     <output url="file://$PROJECT_DIR$/build/classes" />
   </component>
   <component name="ProjectType">
     <option name="id" value="Android" />
   </component>
-</project>
+</project>
```
Comment (rt4914) [inline .idea/misc.xml]: Revert changes from this file. We should not make any change to this file.

**Comment by rt4914** [comment]
@abhinav12k `Talkback` is meant to use `Double-tap in every case`. You should not try to override it to `Single click action`. 

Reason: `Talkback` is used by specially-enabled people. For example, if there is a blind person who is using Oppia with the help of `talkback`, if they want to select some item then single click can be problematic because by mistake they can do single click. But `Double-tap` is something which can only be done intentionally thus making sure that if they want to select any option they they can double tap thus being sure.

**Comment by abhinav12k** [comment]
As per the second point in the checklist (checkbox should not be selected separately), I thought I need to allow the selection of language through a single click that's why I tried to override the talkback actions for that radio button.

**Comment by anandwana001** [comment]
Sorry for making noise here, @abhinav12k please don't change the default checklist. 
Also fix the PR title as given in the default checklist.

**Comment by anandwana001** [comment]
It seems like our PR template has been broken, let's keep this pr as it is, no changes required. 
Just the title - `Fix #2631: A11y Options Language scree`

**Comment by abhinav12k** [comment]
@anandwana001 I made a video of the current behavior should I add that too?

**Comment by anandwana001** [comment]
Yes, you can. @rt4914 will be the main reviewer for this PR, unsubscribing myself.

File: .idea/misc.xml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8"project-jdk-name="1.8" project-jdk-type="JavaSDK">
```
Comment (rt4914) [inline .idea/misc.xml]: This file is still showing changes. Please revert them

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -54,10 +54,12 @@
       android:theme="@style/OppiaThemeWithoutActionBar" />
     <activity
       android:name=".app.options.AppLanguageActivity"
-      android:theme="@style/OppiaThemeWithoutActionBar" />
+      android:theme="@style/OppiaThemeWithoutActionBar"
+      android:label="@string/app_language"/>
```
Comment (rt4914) [inline app/src/main/AndroidManifest.xml]: Please rename this string to `app_language_activity_title`.

Also add ` ` (space) just before `/>`.

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -54,10 +54,12 @@
...
     <activity
       android:name=".app.options.AudioLanguageActivity"
-      android:theme="@style/OppiaThemeWithoutActionBar" />
+      android:theme="@style/OppiaThemeWithoutActionBar"
+      android:label="@string/title_default_audio"/>
```
Comment (rt4914) [inline app/src/main/AndroidManifest.xml]: Please rename this string to `audio_language_activity_title`.

Also add ` ` (space) just before `/>`.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@abhinav12k I have suggested some changes. Along with that I checked your output but this is still not user friendly for A11y users.

Issues that I found.
1. Currently if we traverse though all the languages there is no way the user can know which language is currently selected.
2. Also, once the user changes the language they are not getting any feedback whether the language has been changes or not.

## Possible solution
1. When we traverse to any language currently it outputs `Hindi, 4 of 5`. Instead if we can fix this to give output something like this `checked, Hindi, 4 of 5` (if the item is selected) or `not checked, French, 3 of 5`. 
2. When user double taps to select an option it should output `checked`. 

Note: Do not write any custom code. It should be done in the simplest way.

**Comment by abhinav12k** [comment]
@rt4914 I have updated the video also, please check it out.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@abhinav12k Thanks a lot for this PR. However there are issues with this PR.

1. We are introducing strings which will be output for Talkback. Now generally thats fine but for defaults items and behaviours talkback should be able to handle them directly. Now for example if in future `Talkback` app decides to output `selected` or `not selected` instead of `checked` or `not checked` then because we are setting the strings and not the talkback our app will output `Checked` or `not checked` only and all other apps which have a radio button will output `selected` or `not selected` which will become inconsistent. In general instead of introducing the strings from our end the `Talkback` should handle this.

## Solution
Gist: https://gist.github.com/rt4914/5cd99f88c4d33e8a793f528865d0904b
Output

https://user-images.githubusercontent.com/9396084/109525679-c8c76480-7ad7-11eb-96bd-15f842f1c2dc.mp4

I have basically tried to merge the textview and the radio button. The solution does work but if you apply this solution you will need to fix following things:
1. The UI should not change at all. For this you can check it by manually comparing the screenshots. [Reference](https://www.youtube.com/watch?v=iRzGOj1RWoQ)
2. Because the solution involves removing text view and adding text in the `RadioButton` the test cases will need to be updated too. 

Does this make sense? Feel free to ask questions.

**Comment by abhinav12k** [comment]
@rt4914 I am able to achieve the below output by making some minor changes. Please let me know if it works.

**Updated Output**

https://user-images.githubusercontent.com/51455561/109538097-b3f1cd80-7ae5-11eb-904f-492ce3525059.mp4

**Comment by rt4914** [comment]
> @rt4914 I am able to achieve the below output by making some minor changes. Please let me know if it works.
> 
> **Updated Output**
> 
>  VID_20210301232346.mp4

@abhinav12k In your solution you have updated the strings. But the solution that I am proposing won't even require strings. The `Talkback` will take care of everything.
Just try to do the change that I suggest on some other branch and test the app on Talkback and you will notice that it does say `checked or not checked` without even introducing that string in xml file.

**Comment by abhinav12k** [comment]
@rt4914 the above solution does not require any strings. This output is generated by talkback only.

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -333,6 +333,8 @@
   <string name="app_version_name">App Version %s</string>
   <string name="app_last_update_date">The last update was installed on %s. Use the above version number to send feedback about bugs.</string>
   <!-- OptionsActivity -->
+  <string name="app_language_activity_title">App Language</string>
+  <string name="audio_language_activity_title">Default Audio Language</string>
   <string name="title_reading_text_size">Reading Text Size</string>
   <string name="title_app_language">App Language</string>
   <string name="title_default_audio">Default Audio Language</string>
```
Comment (rt4914) [inline app/src/main/res/values/strings.xml]: You can remover `title_app_language ` and `title_default_audio ` and whereever it was getting used just use `app_language_activity_title ` and `audio_language_activity_title `.

**Review by rt4914 — CHANGES_REQUESTED** [review]
The solution does work nicely.

Please add tests for checking the label. Reference: https://github.com/oppia/oppia-android/blob/ad1eb3dbbd6d6617a9d1cdedc328233171476efe/app/src/sharedTest/java/org/oppia/android/app/topic/TopicActivityTest.kt#L90

Do this for both the activities.

Nice work.

File: app/src/main/res/values/strings.xml (hunk 5 above)
Comment (abhinav12k) [reply app/src/main/res/values/strings.xml]: Done

File: app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt [hunk 6]
```diff
@@ -83,6 +86,11 @@ private const val CHINESE = 3
 @Config(application = AppLanguageFragmentTest.TestApplication::class)
 class AppLanguageFragmentTest {
 
+  @get:Rule
+  var activityTestRule: ActivityTestRule<AppLanguageActivity> = ActivityTestRule(
```
Comment (rt4914) [inline app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt]: Use `val` instead of `var`

File: app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt [hunk 7]
```diff
@@ -104,6 +112,19 @@ class AppLanguageFragmentTest {
...
+  fun testAppLanguageActivity_hasCorrectActivityLabel() {
+    activityTestRule.launchActivity(
+      createAppLanguageActivityIntent(
+        "English"
+      )
+    )
+    val title = activityTestRule.activity.title
+    // Verify that the activity label is correct as a proxy to verify TalkBack will announce the
+    // correct string when it's read out.
+    Truth.assertThat(title).isEqualTo(context.getString(R.string.app_language_activity_title))
```
Comment (rt4914) [inline app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt]: Use direct import and write it as 
`assertThat(title).isEqualTo(context.getString(R.string.app_language_activity_title))`

File: app/src/test/java/org/oppia/android/app/testing/options/AudioLanguageFragmentTest.kt [hunk 8]
```diff
@@ -84,6 +87,11 @@ private const val CHINESE = 4
 @Config(application = AudioLanguageFragmentTest.TestApplication::class)
 class AudioLanguageFragmentTest {
 
+  @get:Rule
+  var activityTestRule: ActivityTestRule<AudioLanguageActivity> = ActivityTestRule(
```
Comment (rt4914) [inline app/src/test/java/org/oppia/android/app/testing/options/AudioLanguageFragmentTest.kt]: Ditto. make it `val`

File: app/src/test/java/org/oppia/android/app/testing/options/AudioLanguageFragmentTest.kt [hunk 9]
```diff
@@ -105,6 +113,19 @@ class AudioLanguageFragmentTest {
...
+  fun testAudioLanguageActivity_hasCorrectActivityLabel() {
+    activityTestRule.launchActivity(
+      createDefaultAudioActivityIntent(
+        "French"
+      )
+    )
+    val title = activityTestRule.activity.title
+    // Verify that the activity label is correct as a proxy to verify TalkBack will announce the
+    // correct string when it's read out.
+    Truth.assertThat(title).isEqualTo(context.getString(R.string.audio_language_activity_title))
```
Comment (rt4914) [inline app/src/test/java/org/oppia/android/app/testing/options/AudioLanguageFragmentTest.kt]: Ditto. 
`assertThat(title).isEqualTo(context.getString(R.string.audio_language_activity_title))`

**Review by rt4914 — CHANGES_REQUESTED** [review]
Suggested some changes.

Also, please do not resolve comments on your own. In practice you should always reply to the comments and the person who has started that comment-thread will resolve the comment.

File: app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt (hunk 6 above)
Comment (abhinav12k) [reply app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt]: Done.
I think you have mentioned me bymistake on a [different PR ](https://github.com/oppia/oppia-android/pull/2839#issuecomment-792609406)

Comment (rt4914) [reply app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt]: @abhinav12k Apologies. Just missed the last characters in user name.

**Comment by rt4914** [comment]
@abhinav12k Do reply to all the comments and mention that its ready for review whenever it is.

File: app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt (hunk 7 above)
Comment (abhinav12k) [reply app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt]: done

File: app/src/test/java/org/oppia/android/app/testing/options/AudioLanguageFragmentTest.kt (hunk 8 above)
Comment (abhinav12k) [reply app/src/test/java/org/oppia/android/app/testing/options/AudioLanguageFragmentTest.kt]: done

File: app/src/test/java/org/oppia/android/app/testing/options/AudioLanguageFragmentTest.kt (hunk 9 above)
Comment (abhinav12k) [reply app/src/test/java/org/oppia/android/app/testing/options/AudioLanguageFragmentTest.kt]: done

**Comment by rt4914** [comment]
@abhinav12k I really like the way you have worked on this PR. It's really impressive.

There is just one edge case missed/incorrectly done in this PR (actually not done incorrectly but something is different for sure.). Before I explain about that, can you please run the tests that you have written and share the screenshot whether they pass or not? Also did you run those tests on Roboelectric or Espresso?

File: app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt [hunk 10]
```diff
@@ -104,6 +112,19 @@ class AppLanguageFragmentTest {
     Intents.release()
   }
 
+  @Test
+  fun testAppLanguageActivity_hasCorrectActivityLabel() {
+    activityTestRule.launchActivity(
```
Comment (anandwana001) [inline app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt]: Could you please add a screenshot showing this test is passing on espresso?

File: app/src/test/java/org/oppia/android/app/testing/options/AudioLanguageFragmentTest.kt [hunk 11]
```diff
@@ -105,6 +113,19 @@ class AudioLanguageFragmentTest {
     Intents.release()
   }
 
+  @Test
+  fun testAudioLanguageActivity_hasCorrectActivityLabel() {
+    activityTestRule.launchActivity(
```
Comment (anandwana001) [inline app/src/test/java/org/oppia/android/app/testing/options/AudioLanguageFragmentTest.kt]: Could you please add a screenshot showing this test is passing on espresso?

**Review by anandwana001 — CHANGES_REQUESTED** [review]

**Comment by abhinav12k** [comment]
Thanks for the appreciation @rt4914 :)
 
I am running the tests using Roboelectric and they are working correctly.
Screenshot_20210308_163627
Screenshot_20210308_163454

**Comment by rt4914** [comment]
> Thanks for the appreciation @rt4914 :)
> 
> I am running the tests using Roboelectric and they are working correctly.
> Screenshot_20210308_163627
> Screenshot_20210308_163454

@abhinav12k Thats great and can you try to run these same tests on espresso too and post results here?

File: app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt (hunk 10 above)
Comment (abhinav12k) [reply app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt]: I haven't done much in testing. Can you please guide me on how can I run these tests on espresso or do I need to write new tests for espresso?

Comment (rt4914) [reply app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt]: @abhinav12k https://github.com/oppia/oppia-android/wiki/Oppia-Android-Testing

**Comment by rt4914** [comment]
@abhinav12k Mostly you won't be able to run the tests on espresso.
**Reason**: Currently your tests are part of a file which is present inside: `https://github.com/oppia/oppia-android/tree/develop/app/src/test/`. All the tests present inside this file are run on `Roboelectric` only.

Instead you should have created new test files and added your test inside `https://github.com/oppia/oppia-android/tree/develop/app/src/sharedTest/java/org/oppia/android/app/` in which the tests run on `Roboelectric` as well as `Espresso`.

If you notice correctly both the paths are quite different and serve different purpose. If you check the reference that I provided for test case thats written in link 2 and no in link 1. https://github.com/oppia/oppia-android/blob/ad1eb3dbbd6d6617a9d1cdedc328233171476efe/app/src/sharedTest/java/org/oppia/android/app/topic/TopicActivityTest.kt#L90


Does this make sense?

[Additional Reference](https://github.com/oppia/oppia-android/wiki/Oppia-Android-Testing)

**Comment by abhinav12k** [comment]
> @abhinav12k Mostly you won't be able to run the tests on espresso.
> **Reason**: Currently your tests are part of a file which is present inside: `https://github.com/oppia/oppia-android/tree/develop/app/src/test/`. All the tests present inside this file are run on `Roboelectric` only.
> 
> Instead you should have created new test files and added your test inside `https://github.com/oppia/oppia-android/tree/develop/app/src/sharedTest/java/org/oppia/android/app/` in which the tests run on `Roboelectric` as well as `Espresso`.
> 
> If you notice correctly both the paths are quite different and serve different purpose. If you check the reference that I provided for test case thats written in link 2 and no in link 1.
> 
> https://github.com/oppia/oppia-android/blob/ad1eb3dbbd6d6617a9d1cdedc328233171476efe/app/src/sharedTest/java/org/oppia/android/app/topic/TopicActivityTest.kt#L90
> 
> Does this make sense?
> 
> [Additional Reference](https://github.com/oppia/oppia-android/wiki/Oppia-Android-Testing)

Thanks for the explanation @rt4914 😀 and sorry about this 😅 actually I wasn't able to see this particular directory from android studio so I  wrote the tests in link 1 assuming the test have to be written there only. I will correct this and then push the changes.

File: app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt (hunk 10 above)
Comment (abhinav12k) [reply app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt]: I have added tests for espresso as well. @rt4914 PTAL

File: app/src/sharedTest/java/org/oppia/android/app/options/AppLanguageActivityTest.kt [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+  lateinit var context: Context
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @After
+  fun tearDown() {
+    Intents.release()
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/options/AppLanguageActivityTest.kt]: releasing intent without initiating is of now use. 

We use `Intents.init()` when we want to send out intent that needs to verify. A common example if you are testing that clicking a button will open a new intent (activity) then `Intents.init()` is required and to clear the intent we need `.release()`. 

So, here in this test file, it looks like there is no use of intent so you can remove this.

File: app/src/sharedTest/java/org/oppia/android/app/options/AppLanguageActivityTest.kt [hunk 13]
```diff
@@ -0,0 +1,159 @@
...
+
+  private fun setUpTestApplicationComponent() {
+    ApplicationProvider.getApplicationContext<TestApplication>().inject(this)
+  }
+
+  @Test
+  fun testAppLanguageActivity_hasCorrectActivityLabel() {
+    activityTestRule.launchActivity(
+      createAppLanguageActivityIntent(
+        "English"
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/options/AppLanguageActivityTest.kt]: add named argument

File: app/src/sharedTest/java/org/oppia/android/app/options/AppLanguageActivityTest.kt [hunk 14]
```diff
@@ -0,0 +1,159 @@
...
+      )
+    )
+    val title = activityTestRule.activity.title
+    // Verify that the activity label is correct as a proxy to verify TalkBack will announce the
+    // correct string when it's read out.
+    assertThat(title).isEqualTo(context.getString(R.string.app_language_activity_title))
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/options/AppLanguageActivityTest.kt:98]: @rt4914 Is changing the language will change the label too? 

Can we have one more test showing that changing the language updating/not updating the label?

File: app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt (hunk 10 above)
Comment (anandwana001) [reply app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt]: Do we require this test here or are we removing this as we have a new file now?

File: app/src/test/java/org/oppia/android/app/testing/options/AudioLanguageFragmentTest.kt [hunk 15]
```diff
@@ -105,6 +113,19 @@ class AudioLanguageFragmentTest {
...
+      createDefaultAudioActivityIntent(
+        "French"
```
Comment (anandwana001) [inline app/src/test/java/org/oppia/android/app/testing/options/AudioLanguageFragmentTest.kt]: add named argument

File: app/src/sharedTest/java/org/oppia/android/app/options/AudioLanguageActivityTest.kt [hunk 16]
```diff
@@ -0,0 +1,158 @@
...
+  lateinit var context: Context
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @After
+  fun tearDown() {
+    Intents.release()
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/options/AudioLanguageActivityTest.kt]: ditto

File: app/src/sharedTest/java/org/oppia/android/app/options/AudioLanguageActivityTest.kt [hunk 17]
```diff
@@ -0,0 +1,158 @@
...
+  }
+
+  private fun setUpTestApplicationComponent() {
+    ApplicationProvider.getApplicationContext<TestApplication>().inject(this)
+  }
+
+  @Test
+  fun testAppLanguageActivity_hasCorrectActivityLabel() {
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/options/AudioLanguageActivityTest.kt]: we are having the same test in `AudioLanguageFragmentTest`, are we removing this file or that added test in file `AudioLanguageFragmentTest`?

**Review by anandwana001 — CHANGES_REQUESTED** [review]
@abhinav12k Did a quick review. PTAL comments.  

@rt4914 Why are we keeping `AppLanguageFragmentTest` Robolectric only? It look like all the simple UI tests?

File: app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt (hunk 10 above)
Comment (abhinav12k) [reply app/src/test/java/org/oppia/android/app/testing/options/AppLanguageFragmentTest.kt]: No we can remove the tests from here, I will remove them.

File: app/src/sharedTest/java/org/oppia/android/app/options/AppLanguageActivityTest.kt (hunk 12 above)
Comment (abhinav12k) [reply app/src/sharedTest/java/org/oppia/android/app/options/AppLanguageActivityTest.kt]: I will remove it.

File: app/src/sharedTest/java/org/oppia/android/app/options/AppLanguageActivityTest.kt (hunk 13 above)
Comment (abhinav12k) [reply app/src/sharedTest/java/org/oppia/android/app/options/AppLanguageActivityTest.kt]: Ok

File: app/src/sharedTest/java/org/oppia/android/app/options/AppLanguageActivityTest.kt (hunk 14 above)
Comment (abhinav12k) [reply app/src/sharedTest/java/org/oppia/android/app/options/AppLanguageActivityTest.kt:98]: No changing the language won't change the activity's label.

File: app/src/test/java/org/oppia/android/app/testing/options/AudioLanguageFragmentTest.kt (hunk 15 above)
Comment (abhinav12k) [reply app/src/test/java/org/oppia/android/app/testing/options/AudioLanguageFragmentTest.kt]: Added

File: app/src/sharedTest/java/org/oppia/android/app/options/AudioLanguageActivityTest.kt (hunk 16 above)
Comment (abhinav12k) [reply app/src/sharedTest/java/org/oppia/android/app/options/AudioLanguageActivityTest.kt]: Removed this

File: app/src/sharedTest/java/org/oppia/android/app/options/AudioLanguageActivityTest.kt (hunk 17 above)
Comment (abhinav12k) [reply app/src/sharedTest/java/org/oppia/android/app/options/AudioLanguageActivityTest.kt]: I have removed the tests from ` app/src/test/java/org/oppia/android/app/testing/options/AudioLanguageFragmentTest.kt` and added them in the ` app/src/sharedTest/java/org/oppia/android/app/options/AudioLanguageActivityTest.kt `

**Comment by abhinav12k** [comment]
@rt4914 I have added a new commit. PTAL

**Comment by rt4914** [comment]
@anandwana001 AppLanguageFragmentTest should not be roboeletric only. It should pass on both. Also, assigning you so that once you approve I can review it. Thanks.

File: app/src/sharedTest/java/org/oppia/android/app/options/AppLanguageActivityTest.kt (hunk 14 above)
Comment (rt4914) [reply app/src/sharedTest/java/org/oppia/android/app/options/AppLanguageActivityTest.kt:98]: @anandwana001 That is not needed at the label is not dependent on the language.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Comment by anandwana001** [comment]
Re-running checks.

**Comment by anandwana001** [comment]
@abhinav12k Could you try running these new test on robolectric once more on local, see if they are passing as nothing has changed much after last wee checked. 

```
 symbol:   class TestAccessibilityModule
  location: class AudioLanguageActivityTest/home/runner/work/oppia-android/oppia-android/app/build/tmp/kapt3/stubs/debugUnitTest/org/oppia/android/app/options/AppLanguageActivityTest.java:12: error: [ComponentProcessor:MiscError] dagger.internal.codegen.ComponentProcessor was unable to process this class because not all of its dependencies could be resolved. Check for compilation errors or a circular dependency with generated code.
public final class AppLanguageActivityTest {
             ^/home/runner/work/oppia-android/oppia-android/app/build/tmp/kapt3/stubs/debugUnitTest/org/oppia/android/app/options/AudioLanguageActivityTest.java:12: error: [ComponentProcessor:MiscError] dagger.internal.codegen.ComponentProcessor was unable to process this class because not all of its dependencies could be resolved. Check for compilation errors or a circular dependency with generated code.
public final class AudioLanguageActivityTest {
             ^/home/runner/work/oppia-android/oppia-android/app/build/tmp/kapt3/stubs/debugUnitTest/org/oppia/android/app/options/AppLanguageActivityTest.java:92: error: [ComponentProcessor:MiscError] dagger.internal.codegen.ComponentProcessor was unable to process this interface because not all of its dependencies could be resolved. Check for compilation errors or a circular dependency with generated code.
    public static abstract interface TestApplicationComponent extends org.oppia.android.app.application.ApplicationComponent {
                           ^/home/runner/work/oppia-android/oppia-android/app/build/tmp/kapt3/stubs/debugUnitTest/org/oppia/android/app/options/AudioLanguageActivityTest.java:92: error: [ComponentProcessor:MiscError] dagger.internal.codegen.ComponentProcessor was unable to process this interface because not all of its dependencies could be resolved. Check for compilation errors or a circular dependency with generated code.
    public static abstract interface TestApplicationComponent extends org.oppia.android.app.application.ApplicationComponent {
```

**Comment by abhinav12k** [comment]
@anandwana001 can you please verify the output? i am getting these outputs after running tests on my local machine
test2
test1

**Comment by rt4914** [comment]
Re-running tests as it should have passed.

**Comment by abhinav12k** [comment]
@anandwana001 PTAL

**Review by anandwana001 — APPROVED** [review]
LGTM, Thanks @abhinav12k

**Comment by abhinav12k** [comment]
I just merged develop branch. Can you please explain me what was the reason for the test failure I am still not able to understand that 😅? @anandwana001

**Comment by anandwana001** [comment]
We have some flaky test, which sometimes fail on GitHub checks.

**Comment by abhinav12k** [comment]
> We have some flaky test, which sometimes fail on GitHub checks.
 
Thanks, @anandwana001

@BenHenning Please review the PR

File: app/src/sharedTest/java/org/oppia/android/app/options/AudioLanguageActivityTest.kt [hunk 18]
```diff
@@ -0,0 +1,153 @@
...
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  private fun setUpTestApplicationComponent() {
+    ApplicationProvider.getApplicationContext<TestApplication>().inject(this)
+  }
+
+  @Test
+  fun testAppLanguageActivity_hasCorrectActivityLabel() {
```
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/android/app/options/AudioLanguageActivityTest.kt]:
```suggestion
  fun testAudioLanguageActivity_hasCorrectActivityLabel() {
```

Small typo in test name.

**Review by BenHenning — COMMENTED** [review]
Thanks @abhinav12k! Just had one comment.

File: app/src/sharedTest/java/org/oppia/android/app/options/AudioLanguageActivityTest.kt (hunk 18 above)
Comment (abhinav12k) [reply app/src/sharedTest/java/org/oppia/android/app/options/AudioLanguageActivityTest.kt]: Added suggestion

**Review by BenHenning — APPROVED** [review]
Thanks @abhinav12k! LGTM.
